sktime/utils/tests/test_mlflow_sktime_model_export.py

```diff
@@ -41,7 +41,12 @@ def mock_s3_bucket():
     -------
     string with name of mock S3 bucket
     """
-    with moto.mock_s3():
+    if moto.__version__ < "5.0.0":
```

Comment:
should we use `_check_soft_depedencies` to ensure we route all version checks through the same entry point?

Reply:
There are so many version checks of this type, so opened a new issue to track together. Merging this PR as you said it's non-blocking, will fix this in a new one.